SConstruct

```diff
@@ -531,141 +535,176 @@ class ObjectBuilder(object):
             if kwds:
                 env = env.Clone()
                 env.Prepend(**kwds)
-            path = UNITY_BUILD_DIRECTORY + filename
-            o = env.Object(Beast.variantFile(path, self.variant_dirs))
+            o = env.Object(Beast.variantFile(filename, self.variant_dirs))
             self.objects.append(o)
 
+def list_sources(base, suffixes):
+    def _iter(base):
```

Comment:
Why pass `base` in and not `suffixes`?  Both of them would be captured by the local function.

Frankly, this isn't a compelling use of yield.  Why not just create an empty list and append to it in line 550?  It's easier, and a tiny bit more efficient (you don't have to create the generator object).

Also, it's bad form to have local variables starting with an `_`. `_` basically means the equivalent of "protected" - but you can't ever see local variables so starting with `_` is fine